datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -390,6 +452,50 @@ impl AsyncFileReader for ParquetFileReader {
     }
 }
 
+impl ParquetFileReaderFactory for DefaultParquetFileReaderFactory {
+    fn create_reader(
+        &self,
+        partition_index: usize,
+        file_meta: FileMeta,
+        metadata_size_hint: Option<usize>,
+        metrics: &ExecutionPlanMetricsSet,
+    ) -> Result<Box<dyn AsyncFileReader + Send>> {
+        let parquet_file_metrics = ParquetFileMetrics::new(
+            partition_index,
+            file_meta.location().as_ref(),
+            metrics,
+        );
+
+        Ok(Box::new(ParquetFileReader {
+            meta: file_meta.object_meta,
+            store: Arc::clone(&self.store),
+            metadata_size_hint,
+            metrics: parquet_file_metrics,
+        }))
+    }
+}
+
+///
+/// BoxedAsyncFileReader has been created to satisfy type requirements of
+/// parquet stream builder constructor.
```

Comment:
Is this necessary after https://github.com/apache/arrow-rs/pull/2366? If not perhaps we can add a reference here so we can eventually clean it up?

Reply:
https://github.com/apache/arrow-rs/pull/2368 is the one that will remove the need for this